src/config/jest-playwright.ts

```diff
@@ -1,42 +1,38 @@
-import path from 'path';
+import path from 'pathe';
 import { getProjectRoot } from 'storybook/internal/common';
 import type { Config } from '@jest/types';
 
-const getTestRunnerPath = () => process.env.STORYBOOK_TEST_RUNNER_PATH ?? '@storybook/test-runner';
+const getTestRunnerPath = () => {
+  return (
+    process.env.STORYBOOK_TEST_RUNNER_PATH ??
+    path.dirname(require.resolve('@storybook/test-runner'))
+  );
+};
 
 /**
  * IMPORTANT NOTE:
  * Depending on the user's project and package manager, it's possible that jest-playwright-preset
  * is going to be located in @storybook/test-runner/node_modules OR in the root node_modules
  *
- * By setting `preset: 'jest-playwright-preset` the change of resolution issues is higher, because
+ * By setting `preset: 'jest-playwright-preset' the change of resolution issues is higher, because
  * the lib might be installed inside of @storybook/test-runner/node_modules but references as if it was
  * in the root node_modules.
  *
- * This function does the same thing as `preset: 'jest-playwright-preset` but makes sure that the
+ * This function does the same thing as `preset: 'jest-playwright-preset' but makes sure that the
  * necessary moving parts are all required within the correct path.
  * */
 const getJestPlaywrightConfig = (): Config.InitialOptions => {
-  const TEST_RUNNER_PATH = getTestRunnerPath();
-  const presetBasePath = path.dirname(
-    require.resolve('jest-playwright-preset', {
-      paths: [path.join(__dirname, '../node_modules')],
-    })
-  );
-  const expectPlaywrightPath = path.dirname(
-    require.resolve('expect-playwright', {
-      paths: [path.join(__dirname, '../node_modules')],
-    })
-  );
+  const resolvedRunnerPath = getTestRunnerPath();
+  const expectPlaywrightPath = path.dirname(require.resolve('expect-playwright'));
   return {
-    runner: path.join(presetBasePath, 'runner.js'),
-    globalSetup: require.resolve(`${TEST_RUNNER_PATH}/playwright/global-setup.js`),
-    globalTeardown: require.resolve(`${TEST_RUNNER_PATH}/playwright/global-teardown.js`),
-    testEnvironment: require.resolve(`${TEST_RUNNER_PATH}/playwright/custom-environment.js`),
+    runner: path.join(resolvedRunnerPath, 'dist/jest-playwright-entries/runner.js'),
+    globalSetup: path.join(resolvedRunnerPath, 'playwright/global-setup.js'),
+    globalTeardown: path.join(resolvedRunnerPath, 'playwright/global-teardown.js'),
```

Comment:
This pathway doesn't seem to account for the fact that globalSetup and globalTeardown are now built.

```suggestion
    globalSetup: path.join(resolvedRunnerPath, 'dist/jest-playwright-entries/setup.js'),
    globalTeardown: path.join(resolvedRunnerPath, 'dist/jest-playwright-entries/teardown.js'),
```

Reply:
Fixed, thanks!